overlord/ifacestate/handlers.go

```diff
@@ -655,12 +657,15 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 			continue
 		}
 
-		ts, err := ConnectOnInstall(st, chg, task, plug.Snap.Name(), plug.Name, slot.Snap.Name(), slot.Name)
-		if err != nil {
-			task.Logf("cannot auto-connect plug %s to %s: %s", connRef.PlugRef, connRef.SlotRef, err)
-			continue
+		if err := checkConnectConflicts(st, chg, plug.Snap.Name(), slot.Snap.Name(), task); err != nil {
+			if _, retry := err.(*state.Retry); retry {
```

Comment:
this feels awkard and is also not correct now I think for the new check that you add below (about "connect" auto-connect tasks we want to wait, not ignore),  I think we need to wait/Retry for all cases except ignore if there as non-ready "auto-connect" task for the other side